lib/Driver/UnixToolChains.cpp

```diff
@@ -278,6 +278,12 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
   if (job.getKind() == LinkKind::Executable && context.OI.SelectedSanitizers) {
     Arguments.push_back(context.Args.MakeArgString(
         "-fsanitize=" + getSanitizerList(context.OI.SelectedSanitizers)));
+
+    // The TSan runtime depends on the blocks runtime and libdispatch.
+    if (context.OI.SelectedSanitizers & SanitizerKind::Thread) {
```

Comment:
Do you want TSan to *require* these libraries, or do you want it to *support them if they're present*? I don't know a whole lot about the upstream work you're integrating here, so the way you're doing it might be correct.

Reply:
The TSan runtime *requires* these libraries, i.e., the linker will error with `undefined symbol` errors if we don't do this.